### DISCUSSION TITLE
Should InfiniteQuery be able to call 'fetchNextPage' when/if enabled = false?

### DISCUSSION THREAD

**Discussion by jsrhodes15** [TanStack/query#2431] [General] [answered]
Hi All

I'm using `useInfiniteQuery` for infinite scrolling on a search feature. The user can enter their filters, and "search". When the user resets the search, the filter is cleared, which in this case causes the `enabled` flag to be false. 

I copy/pasted the `useIntersectionObserver` code from the example in the docs, and had an issue where, when the user reset's the search, the `enabled` flag on the query becomes false, but the `enabled` flag on the Intersection Observer was still true, which called `fetchNextPage` directly. 

This would result in an "empty" search. Now, I fixed the bug in my `useIntersectionObserver` implementation, and all is good. But, I wanted to ask, should this have been allowed to happen on the `react-query` side? Should I be able to call/use `fetchNextPage` even though the query itself is not enabled?

Here is a minimal repro: https://codesandbox.io/s/elegant-fire-nflxm?file=/pages/index.js
☝🏼 just forked the example and made the changes needed to show what I am talking about.

**Comment by jsrhodes15** [comment]
Just want to mention that if this is indeed a bug I would be happy to submit a PR.

**Comment by TkDodo** [comment, marked as answer]
it's an imperative method like `refetch`, which also bypasses the enabled flag by design. I'm not sure if this is intended here, but I don't think it's a severe thing
